tests/test_build_linkcheck.py

```diff
@@ -284,6 +286,36 @@ def test_follows_redirects_on_GET(app, capsys):
         127.0.0.1 - - [] "GET /?redirected=1 HTTP/1.1" 204 -
         """
     )
+    assert warning.getvalue() == ''
+
+
+@pytest.mark.sphinx('linkcheck', testroot='linkcheck-localserver', freshenv=True,
+                    confoverrides={'linkcheck_warn_redirects': True})
+def test_linkcheck_warn_redirects(app, warning):
+    with http_server(make_redirect_handler(support_head=False)):
+        app.build()
+    assert ("index.rst.rst:1: WARNING: redirect  http://localhost:7777/ - with Found to "
+            "http://localhost:7777/?redirected=1\n" in strip_escseq(warning.getvalue()))
```

Comment:
Perhaps verify this is the only warning:
```suggestion
    assert ("index.rst.rst:1: WARNING: redirect  http://localhost:7777/ - with Found to "
            "http://localhost:7777/?redirected=1\n" in strip_escseq(warning.getvalue()))
     assert warning.getvalue().count("\n") == 1

```